cmd/provision/config.go

```diff
@@ -194,11 +194,25 @@ func (p *provision) createPolicySecretData(config *server.Config, verbosef share
 	return err
 }
 
-func (p *provision) createAnalyticsSecretData() error {
-	// load analytics service account credentials
-	cred, err := ioutil.ReadFile(p.analyticsServiceAccount)
-	if err != nil {
-		return err
+// createAnalyticsSecretData creates the analyticsSecretData to be encoded into the config file
+func (p *provision) createAnalyticsSecretData(config *server.Config, verbosef shared.FormatFn) error {
+	var cred []byte
+	// creates the analytics secret if service account is specified
+	if p.analyticsServiceAccount != "" {
+		var err error
+		// load analytics service account credentials
+		cred, err = ioutil.ReadFile(p.analyticsServiceAccount)
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(config.Analytics.CredentialsJSON) > 0 {
+		cred = config.Analytics.CredentialsJSON
```

Comment:
Codecov is reporting this line isn't covered, but this is a key line in the fix... we should probably ensure it's hit.

Reply:
Makes sense.

Reply:
Done.